runtime/src/runtime/fvm.rs

```diff
@@ -335,13 +335,13 @@ where
         })
     }
 
-    fn delete_actor(&self, beneficiary: &Address) -> Result<(), ActorError> {
+    fn delete_actor(&self) -> Result<(), ActorError> {
         if *self.in_transaction.borrow() {
             return Err(
                 actor_error!(assertion_failed; "delete_actor is not allowed during transaction"),
             );
         }
-        Ok(fvm::sself::self_destruct(beneficiary)?)
+        Ok(fvm::sself::self_destruct(false)?)
```

Comment:
Passing `false` here will cause `self_destruct` to fail if there are any remaining funds. At the moment, that's the only behavior we want in the builtin actors.